src/OpenTelemetry.Api/Context/Propagation/BaggagePropagator.cs

```diff
@@ -116,7 +131,8 @@ internal static bool TryExtractBaggage(string[] baggageCollection, out Dictionar
 
             if (string.IsNullOrEmpty(item))
             {
-                continue;
+                baggage = null;
+                return false;
```

Comment:
It feels very strict to ignore the entire baggage collection if any entry is invalid. I'm not so familiar with the baggage spec, but the [tracestate spec](https://www.w3.org/TR/trace-context/#tracestate-header-field-values) specifically calls out that empty list members are allowed:

> Empty and whitespace-only list members are allowed. Vendors MUST accept empty tracestate headers but SHOULD avoid sending them. Empty list members are allowed in tracestate because it is difficult for a vendor to recognize the empty value when multiple tracestate headers are sent. Whitespace characters are allowed for a similar reason, as some vendors automatically inject whitespace after a comma separator, even in the case of an empty header.

Reply:
According to [W3C Baggage spec](https://w3c.github.io/baggage/#definition) empty entries are invalid - entry needs to at least contain a `key` satisfying a [`token`](https://datatracker.ietf.org/doc/html/rfc7230#section-3.2.6) requirement (note at least one char from `tchar` range) followed by a `=`.

Reply:
I see. They are indeed different on whether allowing list-member to be OWS.